### ISSUE TITLE
The "Delegate Type Could Not Be Inferred" when customizing error responses

### ISSUE THREAD

**Issue by belahd** [FastEndpoints/FastEndpoints#845]
I have a build error: 'Delegate Type Could Not Be Inferred' when I try to customize the error response using the following code:

```
app.UseFastEndpoints(x =>
{
    x.Errors.ResponseBuilder = (failures, _, statusCode) => failures.GetProblemDetail();
});
```

**Comment by dj-nitehawk** [comment]
i believe this only pops up with rider + .net 9.
can you try `dotnet build` in a terminal and see if it builds correctly?
also, what happens if you switch the project sdk to .net 8? compiles correctly in rider yes?
if it does, this needs to be reported to jetbrains i'm afraid.

a workaround would be to fully define the delegate like this:

```cs
c.Errors.ResponseBuilder = (List<ValidationFailure> failures, HttpContext ctx, int status) =>
```

**Comment by dj-nitehawk** [comment]
it's a bug in rider: https://youtrack.jetbrains.com/issue/RIDER-119972
should get fixed eventually.

**Comment by dj-nitehawk** [comment]
looks like they fixed it in `2024.3.2`.
i'm no longer getting the error.
